From 65ffbf526ae9007990e7973d1c525e1df2fc20fc Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Tue, 16 Jun 2015 18:49:35 +0800
Subject: [PATCH 0557/5242] MLK-11281-1: input: keyboard: gpio_keys: add
 pinctl set in suspend and resume

commit  f91ea611675082d42d42af616132dc28e71f58a9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add pinctrl restore to support LPSR.

Signed-off-by: Robin Gong <b38343@freescale.com>
(cherry picked from commit f7d40aa30ca2fe337ba9bb8f7eaca4393db900be)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/input/keyboard/gpio_keys.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/input/keyboard/gpio_keys.c b/drivers/input/keyboard/gpio_keys.c
index 052e376..60a56ea 100644
--- a/drivers/input/keyboard/gpio_keys.c
+++ b/drivers/input/keyboard/gpio_keys.c
@@ -988,6 +988,8 @@ static int __maybe_unused gpio_keys_suspend(struct device *dev)
 		mutex_unlock(&input->mutex);
 	}
 
+	pinctrl_pm_select_sleep_state(dev);
+
 	return 0;
 }
 
@@ -997,6 +999,8 @@ static int __maybe_unused gpio_keys_resume(struct device *dev)
 	struct input_dev *input = ddata->input;
 	int error = 0;
 
+	pinctrl_pm_select_default_state(dev);
+
 	if (device_may_wakeup(dev)) {
 		gpio_keys_disable_wakeup(ddata);
 	} else {
-- 
1.7.9.5

